benchmark/bench_dom_api.cpp

```diff
@@ -18,17 +18,15 @@ static void numbers_scan(State& state) {
   dom::parser parser;
   dom::array arr;
   simdjson::error_code error;
-  parser.load(NUMBERS_JSON).get<dom::array>().tie(arr, error);
-  if(error) {
-    cerr << "could not read " << NUMBERS_JSON << " as an array" << endl;
+  if (!parser.load(NUMBERS_JSON).get(arr, error)) {
```

Comment:
If we went the other way (return error), we would have...

```C++
if (error  = parser.load(NUMBERS_JSON).get(arr)) {
}
```

instead of 


```C++
if (!parser.load(NUMBERS_JSON).get(arr, error)) {
}
```

Reply:
I liked this a lot, so I changed it to this :)

You do need double parentheses to deal with a warning about = in conditions:

```c++
if ((error = parser.load(NUMBERS_JSON).get(arr))) {
}
```

I find it just as readable. More so in some ways: the double paren can serve as a kind of marker that you're seeing one of these error = function() constructions.

And I really like the technique of assigning the error. Otherwise you end up with !function() being success and function() meaning an error, which is just not intuitive.